api/maven-api-core/src/main/java/org/apache/maven/api/JavaPathType.java

```diff
@@ -18,6 +18,10 @@
  */
 package org.apache.maven.api;
 
+import javax.tools.DocumentationTool;
```

Comment:
api shouldn't depend on that since it is a generic path setup and there we make it depending on a single consumer so look fishy and not very promishing to be as generic as intended

Reply:
The generic API is the `PathType` interface. This dependency appears in th `JavaPathType` enumeration, which is the specialization for the Java tools.

Reply:
Ok for me if the name reflects it so something like JavaxToolPathType - or whatever makes it explicit. JavaPathType is way more generic IMHO.

Reply:
"Java" here refers to standard Java as defined by Oracle (other entities must use other names, e.g. "Jakarta"). The use of standard `javax.tools` API should not block non-standard Java tools to be added to this `JavaPathType` enumeration if desired, because `Location` is an interface with 2 non-default methods easy to implement (`getName()` and `isOutputLocation()`), and the proposed `JavaPathType.location()` property is `Optional` anyway.

The close connection to `javax.tools` API may actually be desirable because contrarily to other API, `javax.tools` is updated immediately when new Java features requires compiler changes. For example, `javax.tools` has been updated in Java 9 with the addition of new methods specific to modules management. By contrast, Maven and the Plexus compiler did not really reflected that evolution as far as I can see.